experimental/SetPartitions/src/LinearPartition.jl

```diff
@@ -0,0 +1,207 @@
+"""
+    LinearPartition{S <: AbstractPartition, T <: RingElem}
+
+`LinearPartition` represents a linear combination of partitions of type `S`
+with coefficients of type `T`. 
+
+See for example Chapter 5 in [Gro20](@cite) for further information on linear 
+combinations of partitions.
+"""
+struct LinearPartition{S <: AbstractPartition, T <: RingElem}
+    base_ring :: Ring  # parent_type(T)
+    coefficients :: Dict{S, T}
+
+    function LinearPartition{S, T}(ring::Ring, coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElem}
+        @req isconcretetype(S) "Linear combinations are only defined for concrete subtypes of AbstractPartition"
+        @req ring isa parent_type(T) "Ring type is not the parent of the coefficient type"
+        for c in values(coeffs)
+            @req (parent(c) == ring) "Coefficient does not belong to the base ring"
+        end
+        return new(ring, simplify_operation_zero(coeffs))
+    end
+end 
+
+"""
+    linear_partition(ring::Ring, coeffs::Dict{S, <: Any}) where {S <: AbstractPartition}
+
+Construct a linear partition over `ring` with coefficients `coeffs`.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d") 
+(Univariate polynomial ring in d over QQ, d)
+
+julia> linear_partition(S, Dict(set_partition([1, 2], [1, 1]) => 4, set_partition([1, 1], [1, 1]) => 4*d))
+LinearPartition{SetPartition, QQPolyRingElem}(Univariate polynomial ring in d over QQ, Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 2], [1, 1]) => 4, SetPartition([1, 1], [1, 1]) => 4*d))
+```
+"""
+function linear_partition(ring::Ring, coeffs::Dict{S, <: Any}) where {S <: AbstractPartition}
+    ring_coeffs = Dict(p => ring(c) for (p, c) in coeffs)
+    return LinearPartition{S, elem_type(ring)}(ring, ring_coeffs)
+end
+
+"""
+    linear_partition(ring::Ring, terms::Vector{Tuple{S, <: Any}}) where {S <: AbstractPartition}
+
+Return a `LinearPartition` generated from the Vector `term` of 2-tuples,
+where the first element in the tuple is a `RingElement` and the second
+a `SetPartition`. Furthermore simplify the term before initializing
```

Comment:
that is not true. from the signature, the first element in the tuple is a AbstractPartition.

Furthermore, you should mention `ring` in the docstring, and that the second elements of the tuples get tried to get coerced into it